jigglypuffRL/common/utils.py

```diff
@@ -56,24 +58,67 @@ def evaluate(algo, num_timesteps=1000):
     print("Average Reward: {}".format(total_reward / num_timesteps))
 
 
-def save_params(algo):
-    if algo.save_version is None:
-        torch.save(algo.checkpoint, "{}.pt".format(algo.save_name))
+def save_params(algo, timestep):
+    algo_name = algo.__class__.__name__
+    env_name = algo.env.unwrapped.spec.id
+    directory = algo.save_model
+    path = "{}/{}_{}".format(
+        directory, algo_name, env_name
+    )
+
+    if timestep == algo.save_interval:
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+            log = {}
+            run_num = 0
+        else:
+            f = open("{}/log.pkl".format(directory), "rb")
+            log = pickle.load(f)
+            f.close()
+            run_num = log[path] + 1
+
+        if not os.path.exists(path):
+            os.makedirs(path)
     else:
-        torch.save(
-            algo.checkpoint, "{}-{}.pt".format(
-                algo.save_name, algo.save_version
-            )
-        )
+        f = open("{}/log.pkl".format(directory), "rb")
+        log = pickle.load(f)
+        f.close()
+        run_num = log[path]
```

Comment:
Is there a way to combine the two if conditions?

Reply:
I don't think so, the first if is run for the first time the model is being saved. The run_num needs to be set to 0 or updated to a new value. The second if is for all other saves during the training process(not a separate run)

Reply:
So this is not taking care of separate runs right? Also, I'm not sure if I'm getting my point across here.

If you have a timestep already in the save name, we don't really need the `run_id` to be changing in that particular run.

The `run_id` should really keep track of multiple `different` runs. I think I'd mentioned a way to do this in a previous comment. You should be checking the directory for last run and add 1 to its run id

Reply:
Yeah, this is precisely what it's doing. If you run the py file once, it creates multiple different pt files with the same run_num but at different timesteps. These pt files are all saved under a directory named something like DDPG_CartPole-v0. The log.pkl file is a dictionary that saves the latest run_num and latest timestep given a specific run_num. Run_num is an optional variable which can be set differently when initially loaded(if not chosen, the latest one will be loaded). As for timestep, I assumed only the latest timestep would be used for practical purposes except for debugging. Let me know if it's clear now.

Reply:
Ok, this sounds good